application/controllers/EventRulesController.php

```diff
@@ -45,6 +45,7 @@ public function init()
     public function indexAction(): void
     {
         $eventRules = Rule::on(Database::get());
+        $this->sessionNamespace->delete('-1');
```

Comment:
Note that, as I just saw that the `New Event Rule` button used to have a parameter with such functionality, you should use a separate action for new event rules. This then should clear the session before writing anything to it upon submit.

Reply:
/And now I noticed that this very action is already the `addAction` here, but that's used to set up a new rule. i.e. It's not just the endpoint for the very first modal, it's a detail.

I think that there doesn't need to be a distinction between `event-rule?id=-1` and `event-rule?id!=-1`. The `event-rule/index` route should be able to handle new rules (=-1) and existing ones (!=-1). There's not much difference between them, only a few buttons and event handlers. By transferring the responsibility to create buttons to the form, this is already not part of the action anymore. So that only leaves the events and those are just not emitted in case of a new rule, so it doesn't really hurt that they are defined.

Reply:
So in the end:

`event-rules/add` is the endpoint for the the `New Event Rule` button, this resets the session upon submit
`event-rule/index` is the endpoint to configure new and existing rules
`event-rule/edit` is the endpoint to edit new and existing rules